pkg/controller/controller_instance.go

```diff
@@ -277,13 +277,14 @@ func (c *controller) reconcileServiceInstanceDelete(instance *v1beta1.ServiceIns
 		return nil
 	}
 
-	// If there is no op in progress, and the instance either was never
-	// provisioned or was already deprovisioned due to orphan mitigation,
+	// If there is no op in progress, and there was never a provision request
+	// sent for the instance, or the instance has already been deprovisioned,
 	// we can just clear the finalizer and delete. One possible scenario
-	// is if the service class name referenced never existed.
+	// is if the service class name referenced never existed. Another scenario
+	// is if the instance was deleted via orphan mitigation.
 	if instance.Status.CurrentOperation == "" &&
```

Comment:
I think @n3wscott should take a look at this, because he's working on a closely related bug that will likely touch this code.

Reply:
We do not want to clear the finalizer here if the deprovision operation failed